docs/dev/reports/README.md

```diff
@@ -0,0 +1,469 @@
+
+## Report types
+
+In the following sections we will dicuss the different report types, how they are managed and how Kyverno uses them.
+
+### Admission reports
+
+Admission reports are created at resource admission time.
+Every time a resource is being admitted, the webhook looks up the policies that need to be evaluated against the resource and a report is created containing the results for this particular admission request.
+
+```mermaid
+sequenceDiagram
+Webhook->>Policy cache: Lookup policies for admitted resource
+loop Evaluation
+    Webhook->>Engine: Evaluate policies against admitted resource
+    Engine-->>Webhook: Collect and aggregate results
+end
+Webhook->>Report: Create admission report
+```
+
+It's important to note that a resource can be admitted multiple times, CREATE/UPDATE/DELETE, etc... the exact same resource can be admitted multiple times and it's perfectly fine.
+
+One consequence of this is that Kyverno can produce a lot of admission reports for the same resource.
+
+Keeping all those admission reports would be highly inefficient and they would accumulate in the cluster as new admission reports come in.
+To prevent admission reports from piling up in the cluster, Kyverno contiuously aggregates admission reports into a single one per underlying resource.
+
+```mermaid
+flowchart LR
+A1(Admission report - ephemeral) --> B(Admission reports controller)
+A2(Admission report - ephemeral) --> B(Admission reports controller)
+A3(Admission report - ephemeral) --> B(Admission reports controller)
+A4(Admission report - ephemeral) --> B(Admission reports controller)
+B(Admission reports controller) --> AGG1(Aggregate ephemeral admission reports)
+AGG1(Aggregate ephemeral reports) -->|create/update| AGG2(Admission report - aggregated)
+B(Admission reports controller) -->|delete| D1(Admission report - ephemeral)
+B(Admission reports controller) -->|delete| D2(Admission report - ephemeral)
+B(Admission reports controller) -->|delete| D3(Admission report - ephemeral)
+B(Admission reports controller) -->|delete| D4(Admission report - ephemeral)
+```
+
+With that in mind, it's important to note that most admission reports should be short lived and exist only until they have been aggregated.
+Aggregated admission reports can be identified by the `audit.kyverno.io/report.aggregate` label.
+
+Only the aggregated report is expected to be long lived and evtually share the same lifecycle as the underlying resource.
+To aggergate reports together, Kyverno groups results per policy/rule and keeps the most recent one.
+
+### Backgroud scan reports
+
+Background scan reports are produced in the background at regular intervals or when something changes in the policy or the resource.
+There is always a one-to-one relationship between a resource and a background scan report.
+
+Kyverno updates the same background scan report over and over when policies are evaluated against a resource in the background scan process.
```

Comment:
Why not aggregate to the admission report i.e. combine admission and background reports?